compiler/extensions/cpp/runtime/src/zserio/UrlClient.h

```diff
@@ -0,0 +1,35 @@
+#ifndef ZSERIO_URLCLIENT_H_INC
+#define ZSERIO_URLCLIENT_H_INC
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace zserio
+{
+
+class UrlClient
+{
+    struct Impl;
+
+public:
+    UrlClient();
+    ~UrlClient();
+
+    std::vector<uint8_t> fetchResource(const std::string& uri);
+
+    enum Status {
+        Healthy,
```

Comment:
We may think about having a common Status enum/struct/class for all kind of communication services or we just also choose to use exceptions for reporting errors as the rest of the zserio runtime does.